R/methods-MSnExp.R

```diff
@@ -565,3 +565,15 @@ setAs("MSnExp", "data.frame", function(from) {
 })
 as.data.frame.MSnExp <- function(x, row.names = NULL, optional=FALSE, ...)
     as(x, "data.frame")
+
+setAs("MSnExp", "Spectra", function(from) {
+    fdta <- fData(from)
+    red_cn <- c("fileIdx", "spIdx", "smoothed", "seqNum", "acquisitionNum",
+                "msLevel", "polarity", "originalPeaksCount", "totIonCurrent",
+                "retentionTime", "basePeakMZ", "basePeakIntensity",
+                "collisionEnergy", "ionisationEnergy", "precursorScanNum",
+                "precursorMZ", "precursorCharge", "precursorIntensity",
+                "mergedScan", "centroided", "spectrum")
+    fdta <- fdta[, !colnames(fdta) %in% red_cn, drop = FALSE]
```

Comment:
Why aren't all feature variables passed along during coercion? I would prefer to keep all of them by default and have a `selectFeatureData` function to select those to keep.

Reply:
Indeed, thank you. Is there already an accessor for these (`centroided,Spectra`, ...)? The implementation will be identical to the corresponding in memory `MSnExp` methods.

Reply:
Yes, all methods to access slot values for `Spectrum` objects are also implemented for `Spectra` objects (in *methods-Spectra.R*), returning always a vector of length equal to `Spectra` with the corresponding slot values.